hail/src/main/scala/is/hail/expr/types/physical/PSubsetStruct.scala

```diff
@@ -0,0 +1,225 @@
+
+  override def setFieldMissing(structAddress: Code[Long], fieldIdx: Int): Code[Unit] =
+    ps.setFieldMissing(structAddress, idxMap(fieldIdx))
+
+  override def setFieldPresent(structAddress: Long, fieldIdx: Int): Unit =
+    ps.setFieldPresent(structAddress, idxMap(fieldIdx))
+
+  override def setFieldPresent(structAddress: Code[Long], fieldIdx: Int): Code[Unit] =
+    ps.setFieldPresent(structAddress, idxMap(fieldIdx))
+
+  override def fieldOffset(structAddress: Long, fieldIdx: Int): Long =
+    ps.fieldOffset(structAddress, idxMap(fieldIdx))
+
+  override def fieldOffset(structAddress: Code[Long], fieldIdx: Int): Code[Long] =
+    ps.fieldOffset(structAddress, idxMap(fieldIdx))
+
+  override def loadField(structAddress: Long, fieldIdx: Int): Long =
+    ps.loadField(structAddress, idxMap(fieldIdx))
+
+  override def loadField(structAddress: Code[Long], fieldIdx: Int): Code[Long] =
+    ps.loadField(structAddress, idxMap(fieldIdx))
+
+  // FIXME: goal is to ensure isn't constructed
+  // byteSize, alignment needed in InferPType...
+  // FIXME: the correct answer here depends on our structFundamentalType decision
+  override val byteSize = ps.byteSize
+  override val alignment = ps.alignment
+
+  override def appendKey(key: String, sig: PType): PStruct = ???
+
+  def insertFields(fieldsToInsert: TraversableOnce[(String, PType)]): PSubsetStruct = ???
+
+  override def initialize(structAddress: Long, setMissing: Boolean): Unit = ???
+
+  override def stagedInitialize(structAddress: Code[Long], setMissing: Boolean): Code[Unit] = ???
+
+  override def allocate(region: Region): Long = ???
+
+  override def allocate(region: Code[Region]): Code[Long] = ???
+
+  // This would mean changing the requiredeness of the backing PStruct, since requiredeness in the view
+  // is derived from the backign PStruct
+  override def setRequired(required: Boolean): PType = ???
+
+  override def copyFromType(mb: EmitMethodBuilder[_], region: Value[Region], srcPType: PType, srcAddress: Code[Long], deepCopy: Boolean): Code[Long] = ???
+
+  override def copyFromTypeAndStackValue(mb: EmitMethodBuilder[_], region: Value[Region], srcPType: PType, stackValue: Code[_], deepCopy: Boolean): Code[_] = ???
+
+  override protected def _copyFromAddress(region: Region, srcPType: PType, srcAddress: Long, deepCopy: Boolean): Long = ???
+
+  override def constructAtAddress(mb: EmitMethodBuilder[_], addr: Code[Long], region: Value[Region], srcPType: PType, srcAddress: Code[Long], deepCopy: Boolean): Code[Unit] = ???
+
+  override def constructAtAddress(addr: Long, region: Region, srcPType: PType, srcAddress: Long, deepCopy: Boolean): Unit = ???
+}
+
+object PSubsetStructSettable {
+  def apply(cb: EmitCodeBuilder, pt: PSubsetStruct, name: String, sb: SettableBuilder): PSubsetStructSettable = {
+    new PSubsetStructSettable(pt, sb.newSettable(name))
+  }
+}
+
+class PSubsetStructSettable(val pt: PSubsetStruct, val a: Settable[Long]) extends PBaseStructValue with PSettable {
```

Comment:
This should have the same structure as the type -- it should take a PBaseStructValue and the type.